tests/test_dockerfile.py

```
@@ -0,0 +1,143 @@
+# Test framework: pytest
+# Purpose: Validate Dockerfile instructions introduced in the PR diff.
+# Focus areas: base image, WORKDIR, dependency install, COPY order, ENV flags.
+
+import re
+from pathlib import Path
+import pytest
+import logging
+from typing import Callable, Optional
```

Ruff (0.12.2)

Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)